### ISSUE TITLE
Does Gadgetbridge use less ram than proprietary apps? 

### ISSUE THREAD

**Issue by ShapeShifter499** [Freeyourgadget/Gadgetbridge#436]
This is not a issue, the app works fine for me so far, I just have a question and didn't know where else to ask. Has anyone determined if Gadgetbridge uses less or more ram than the proprietary apps such as the ones from Pebble and Xiaomi?

**Comment by cpfeiffer** [comment]
I don't think anyone of us developers has measured it, given that we actually do not run the proprietary apps. Are you short on RAM or are you just interested?

**Comment by ShapeShifter499** [comment]
I'm just generally curious, I've been trimming apps that slow down my device and might hinder privacy. Swapping out services for others that take less resources and all that good stuff. When I ran this on a 1GB Ram Moto G 1st Gen it seemed to slow things down quite a bit. Now I have a Moto X Pure with 3GB of Ram and I don't notice many issues.

**Comment by cpfeiffer** [comment]
The only place where we use a bit more RAM than really necessary is the charts (display of activity data). We have no external server which aggregates data for us, so we do it ourselves, or, to be honest, at the moment, we analyze the data in real-time.

If you notice that we use too much RAM when displaying activity, we could fix that by storing the analysis results in the db for later use.

Everything else should be pretty good, in terms of CPU, RAM and battery usage.

**Comment by cpfeiffer** [comment]
But if you notice anything unexpected, please tell us.

**Comment by danielegobbetti** [comment]
@ShapeShifter499 when you say:
> it seemed to slow things down quite a bit
you mean that the phone felt generally more sluggish or that launching/interacting with GB took a long time?

I ask because there are two main components (that then break down to many more): 
- the background service to make android and the gadget to interoperate
- the GB user-facing app

**Comment by Almtesh** [comment]
I think that RAM used by proprietary apps on Android is a little obscure since they often use a lot of services on Android and it's hardly measurable.
Apps like GB are more since they are more independant, so, that's not easily measurable, in my opinion.

**Comment by danielegobbetti** [comment]
While developing #445 I took an eye on the monitors provided by AS, the basic functionalities (keeping the connection open, sending notifications, etc.) allocate ~8MB of RAM on my device. Of course the RAM usage can increase (tenfold or more) when viewing complex configuration pages for pebble watchapps (local webview). I strongly suggest we close this issue.

**Comment by ShapeShifter499** [comment]
Sorry for not commenting, was a bit busy. 

@danielegobbetti 
  < "you mean that the phone felt generally more sluggish or that launching/interacting with GB took a long time?"

The phone was sluggish but seemed a bit better when I disabled gadgetbridge. Also while using gadgetbridge my Pebble time watch and the watchface I had kept letting me know it was disconnecting and reconnecting.   I do add that it could have been just the particular phone I had (and still have), LG Volt which is a 1.2Ghz quad core and has only 1GB ram.  

Now I use a more beefier phone, a Moto X Pure with 3GB ram and hexa-core (2 big and 4 small cores).  The above do not apply to my use case any more. But anything that can help with performance and speed would be good.
